.gitmodules

```diff
@@ -23,3 +23,9 @@
 [submodule "src/3rd_party/onnxjs"]
 	path = src/3rd_party/onnxjs
 	url = https://github.com/abhi-agg/onnxjs.git
+[submodule "src/3rd_party/simde-no-tests"]
+	path = src/3rd_party/simde-no-tests
+	url = https://github.com/simd-everywhere/simde-no-tests/
```

Comment:
Why are we doing this instead of https://github.com/JishinMaster/simd_utils/blob/master/neon_mathfun.h

Reply:
For the record, SIMDE was recommended by @XapaJIaMnu. After a second pass at source, there's something weird going on at https://github.com/marian-nmt/marian-dev/blob/23c36ec1a3c71cc75bc49fd3e39a4b1d8636589d/src/common/types.h#L20. 

`float32x8_t` is added on AVX everything else is missing? https://github.com/marian-nmt/marian-dev/blob/23c36ec1a3c71cc75bc49fd3e39a4b1d8636589d/src/common/types.h#L201-L232

Reply:
There's a few other places where manual sse/avx/avx2 code paths are used outside of mathisfun. One of them is the transpose, another one had something to do with the output layer.

Reply:
ARM has 128-bit SIMD.  The AVX stuff you refer to is an optional expansion to 256-bit SIMD for processors that support that width, which ARM does not.

Reply:
[`ProdBias`](https://github.com/browsermt/marian-dev/blob/199201eb89b2941afdadb14164e936d412f897ad/src/tensors/cpu/prod.cpp#L203-L213) is only defined, unable to find usage. `ProdBias` is the only place where [`AddBias](https://github.com/browsermt/marian-dev/blob/199201eb89b2941afdadb14164e936d412f897ad/src/tensors/cpu/integer_common.cpp#L7) is used. `AddBias` requires intrinsics. As this is not used, unable to test. What are the implications of removing this?

Reply:
Transpose locations:
1. https://github.com/browsermt/marian-dev/blob/199201eb89b2941afdadb14164e936d412f897ad/src/tensors/cpu/tensor_operators.cpp#L292
2. https://github.com/browsermt/marian-dev/blob/199201eb89b2941afdadb14164e936d412f897ad/src/tensors/cpu/expression_graph_packable.h#L8

https://github.com/jerinphilip/marian/pull/8 (stacked behind this branch) is an attempt to remove SIMDE. A few edits in `types.h` reveal compile errors in areas like https://github.com/browsermt/marian-dev/blob/199201eb89b2941afdadb14164e936d412f897ad/src/functional/operators.h#L268-L280

There's sse_mathfun being used for the other functions, neon_mathfun can probably be dropped in instead. Is this the output layer bits being discussed above?

A lot of what I'm doing looks like something an emulation library (SIMDE) should already be doing. Any suggestions to measure the improvements?